In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.ensemble import RandomForestRegressor
from sklearn import datasets
boston = datasets.load_boston()
bos = pd.DataFrame(boston.data, columns=boston.feature_names)
bos['price'] = boston.target

In [2]:
bos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 506 entries, 0 to 505
Data columns (total 14 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   CRIM     506 non-null    float64
 1   ZN       506 non-null    float64
 2   INDUS    506 non-null    float64
 3   CHAS     506 non-null    float64
 4   NOX      506 non-null    float64
 5   RM       506 non-null    float64
 6   AGE      506 non-null    float64
 7   DIS      506 non-null    float64
 8   RAD      506 non-null    float64
 9   TAX      506 non-null    float64
 10  PTRATIO  506 non-null    float64
 11  B        506 non-null    float64
 12  LSTAT    506 non-null    float64
 13  price    506 non-null    float64
dtypes: float64(14)
memory usage: 55.5 KB


In [3]:
bos.head(5)

CRIM    ZN  INDUS  CHAS    NOX     RM   AGE     DIS  RAD    TAX  \
0  0.00632  18.0   2.31   0.0  0.538  6.575  65.2  4.0900  1.0  296.0   
1  0.02731   0.0   7.07   0.0  0.469  6.421  78.9  4.9671  2.0  242.0   
2  0.02729   0.0   7.07   0.0  0.469  7.185  61.1  4.9671  2.0  242.0   
3  0.03237   0.0   2.18   0.0  0.458  6.998  45.8  6.0622  3.0  222.0   
4  0.06905   0.0   2.18   0.0  0.458  7.147  54.2  6.0622  3.0  222.0   

   PTRATIO       B  LSTAT  price  
0     15.3  396.90   4.98   24.0  
1     17.8  396.90   9.14   21.6  
2     17.8  392.83   4.03   34.7  
3     18.7  394.63   2.94   33.4  
4     18.7  396.90   5.33   36.2

In [4]:
X = bos.iloc[:,0:-1]
y = bos.iloc[:,-1]

In [5]:
from sklearn.model_selection import KFold
cv = KFold(n_splits=10, shuffle = True, random_state=12)

In [6]:
regressor = RandomForestRegressor(n_estimators = 10, random_state = 42)
for train_fold, valid_fold in cv.split(X):
    train = X.loc[train_fold]
    valid = X.loc[valid_fold] 
    
    train_y = y.loc[train_fold]
    valid_y = y.loc[valid_fold]
    
    model = regressor.fit(X = train, y = train_y)
    model_pred = model.predict(X = valid)
    errors = abs(model_pred - valid_y)

In [7]:
print('Mean Absolute Error:', round(np.mean(errors), 3))

Mean Absolute Error: 2.417


In [8]:
# Splitting the dataset into Training set and Test set
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state = 42)

In [9]:
regressor = RandomForestRegressor(n_estimators = 10, random_state = 42)
regressor.fit(X_train,y_train)

RandomForestRegressor(n_estimators=10, random_state=42)

In [10]:
y_pred = regressor.predict(X_test)

In [11]:
# Print the first 5 true responses
y_test[0:5].values

array([23.6, 32.4, 13.6, 22.8, 16.1])

In [12]:
# Print the first 5 Predicted responses
y_pred[0:10]

array([23.49, 30.14, 17.86, 23.32, 16.29, 21.8 , 20.85, 16.54, 20.75,
       19.96])

In [13]:
# Calculate the errors
errors = abs(y_pred - y_test)

In [14]:
# Print out the mean absolute error
print('Mean Absolute Error:', round(np.mean(errors), 3))

Mean Absolute Error: 2.077


When the 10 fold cross validation is applied on datapoints those fit in randomforest regressor Mean absolute error value is 2.417

Mean absolute error value of 2.077 while train test split applied in random forest regressor.